ansible_collections/arista/avd/roles/eos_designs/templates/facts/switch/switch.j2

```diff
@@ -146,9 +149,39 @@ switch:
 {# switch.router_id #}
   router_id: {% include switch.ip_addressing.router_id %}
 
+{%     if design.type == "mpls-vpn" and underlay_ipv6 is arista.avd.defined(true) %}
+{%         if switch_data.combined.loopback_ipv6_pool is arista.avd.defined(fail_action='error',var_name='loopback_ipv6_pool') %}
+{# IPv6 related facts #}
+{# switch.loopback_ipv6_pool #}
+{%             set loopback_ipv6_pool = switch_data.combined.loopback_ipv6_pool | arista.avd.default() %}
+  loopback_ipv6_pool: {{ loopback_ipv4_pool }}
+
+{# switch.loopback_ipv4_offset #}
+{%             set loopback_ipv6_offset = switch_data.combined.loopback_ipv6_offset | arista.avd.default(0) %}
+  loopback_ipv6_offset: {{ loopback_ipv6_offset }}
+
+{# switch.loopback_ipv4_description #}
+  loopback_ipv6_description: {{ switch_data.combined.loopback_ipv6_description | arista.avd.default(overlay_loopback_description) }}
+
+{# switch.ipv6_router_id #}
+  ipv6_router_id: {% include switch.ip_addressing.ipv6_router_id %}
+{%         endif %}
+{%     endif %}
+
+{# switch.evpn_role #}
+{%     set switch_evpn_role = switch_data.combined.evpn_role | arista.avd.default(
+                          switch.default_evpn_role) %}
+  evpn_role: {{ switch_evpn_role }}
+
 {# switch.bgp_as #}
 {%     if bgp_as is arista.avd.defined and switch_overlay_routing_protocol == 'ibgp' %}
+{%         if design.type == "mpls-vpn" %}
```

Comment:
Try to redesign the logic to only have one `bgp_as: "{{ bgp_as }}"`.
It is not very clear why you test for mpls-vpn _and_ evpn_role, and only configure bgp_as if evpn is also turned on.

Reply:
Fixed.